Tasks/PublishCodeCoverageResultsV1/cieventlogger.ts

```diff
@@ -0,0 +1,51 @@
+import * as tl from 'azure-pipelines-task-lib/task';
+
+const area: string = 'CodeCoverage';
+const feature: string = 'PublishCodeCoverageResultsV1';
+const consolidatedCiData: { [key: string]: any; } = <{ [key: string]: any; }>{};
+
+function getDefaultProps() {
+    return {
+        releaseuri: tl.getVariable('Release.ReleaseUri'),
+        releaseid: tl.getVariable('Release.ReleaseId'),
+        builduri: tl.getVariable('Build.BuildUri'),
+        buildid: tl.getVariable('Build.Buildid'),
+        osType: tl.osType()
+    };
+}
+
+export function addToConsolidatedCi(key: string, value: any) {
+    consolidatedCiData[key] = value;
```

Comment:
it is unlikely that we run into a NRE here, From what I lookedup, if we pass a null or undefined here, it will simply be coerced a 'null' or 'undefined' string